spark/src/main/scala/org/apache/iceberg/spark/SparkTableUtil.scala

```diff
@@ -297,5 +306,104 @@ object SparkTableUtil {
       )
     }
   }
+
+  private def buildManifest(conf: SerializableConfiguration,
+      sparkDataFiles: Seq[SparkDataFile],
+      partitionSpec: PartitionSpec,
+      basePath: String): Iterator[Manifest] = {
+    if (sparkDataFiles.isEmpty) {
+      Seq.empty.iterator
+    }
```

Comment:
This check will not return empty iterator directly. If you want a return here you should change to `return Seq.empty.iterator` (this is not scalastyle).

Reply:
Good catch. I assumed this method had not been changed much. This should be fixed by wrapping the write in an else case.